frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NpmRunner.java

```diff
@@ -37,6 +37,7 @@ public DefaultNpmRunner(NodeExecutorConfig config, ProxyConfig proxyConfig, Stri
 
             arguments.add("--https-proxy=" + proxy.getUri().toString());
             arguments.add("--proxy=" + proxy.getUri().toString());
+            arguments.add("--noproxy=" + proxy.getNonProxyHosts());
```

Comment:
To handle cases where nonProxyHosts is empty or absent perhaps this needs a null/empty check to avoid `--noproxy=null` or `noproxy=  ` ?

Reply:
I added the check, and I added a unit test for that check (as well as the entire `buildArguments` method)